src/azure-cli/azure/cli/command_modules/appconfig/_utils.py

```diff
@@ -118,3 +125,38 @@ def prep_null_label_for_url_encoding(label=None):
         label = '"{0}"'.format(label)
         label = ast.literal_eval(label)
     return label
+
+
+def get_appconfig_data_client(cmd, name, connection_string, auth_mode, endpoint):
+    azconfig_client = None
+    if auth_mode == "key":
+        connection_string = resolve_connection_string(cmd, name, connection_string)
+        try:
+            azconfig_client = AzureAppConfigurationClient.from_connection_string(connection_string=connection_string,
+                                                                                 user_agent=HttpHeaders.USER_AGENT)
+        except ValueError as ex:
+            raise CLIError("Operation failed to due to an exception: {}".format(str(ex)))
+
+    if auth_mode == "login":
+        if not endpoint:
+            try:
+                if name:
+                    _, endpoint = resolve_store_metadata(cmd, name)
+                else:
+                    raise CLIError("App Configuration endpoint or name should be provided if auth mode is 'login'.")
+            except Exception as ex:
+                raise CLIError(str(ex) + "\nYou may be able to resolve this issue by providing App Configuration endpoint instead of name.")
+
+        from azure.cli.core._profile import Profile
+        profile = Profile(cli_ctx=cmd.cli_ctx)
+        # Due to this bug in get_login_credentials: https://github.com/Azure/azure-cli/issues/15179,
+        # we need to manage the AAD scope by passing appconfig endpoint as resource
+        cred, _, _ = profile.get_login_credentials(resource=endpoint)
+        try:
+            azconfig_client = AzureAppConfigurationClient(credential=cred,
+                                                          base_url=endpoint,
+                                                          user_agent=HttpHeaders.USER_AGENT)
+        except (ValueError, TypeError) as ex:
+            raise CLIError("Operation failed to due to an exception: {}".format(str(ex)))
```

Comment:
For the error message, I would recommend using a more specific word, rather than the generic "Operation" to help you troubleshoot the issue, something like:

> Failed to initialize AzureAppConfigurationClient due to an exception:

Reply:
Done